corehq/apps/users/models.py

```diff
@@ -2423,6 +2423,11 @@ def to_ota_restore_user(self, domain, request_user=None):
             request_user=request_user
         )
 
+    def get_owner_ids(self, domain):
+        owner_ids = [self.user_id]
+        owner_ids.extend(loc.location_id for loc in self.get_case_owning_locations(domain))
+        return owner_ids
```

Comment:
That's what I was hoping for originally, but actually in `CommCareUser` the operative line is
```python
        owner_ids.extend([g._id for g in self.get_case_sharing_groups()])
```
that is, it uses `get_case_sharing_groups` instead of `get_case_owning_locations`.  Locations are one type of case sharing group, but there are others, at present only applicable to mobile workers.

`get_case_sharing_groups` is also used in the restore to get pseudo group objects that I'm not sure are actually useful for anything (maybe some case sharing app workflows?), but they're in there.  Unfortunately, that one needs access to the full location object, or at least multiple fields from it, whereas for `get_owner_ids` I only need the ID.  I was hoping to implement `get_case_owning_locations` in such a way that it'd return a queryset, so I could use `.values_list('location_id', flat=True)` here and `.all()` there, while sharing the same logic.  I wasn't able to get that working though - ran into errors making a union between the two `yield from` parts of that function.

Reply:
The pseudo groups have the IDs set to the location ID anyways, so it's the same.  For `CommCareUser` we get a list of locations, convert that to groups with the same IDs, then grab just the IDs of those groups for this method.  It's pretty roundabout, except that we do also need the groups themselves to generate the restore file.

For CCUsers, locations are used in a few ways in the restore:

* Directly assigned locations are available in user data (in the `Registration` element)
* There is a location fixture present in the restore
* case-owning locations are converted to groups and included in the groups fixture
* case-owning locations' IDs are sent to the case restore logic (via this groups mechanism)

What I'm doing here is circumventing the groups mechanism, and thus dropping bullet 3 there.  If the user is in exactly one case-sharing group, you can build an app that selects that group automatically.  The group can be a location or a regular group, but it breaks if there's more than one.  This workflow won't work for web users with the changes I've made here, but I think projects generally assign case owners more explicitly anyways, for instance by using the `commcare_primary_case_sharing_id` field in user data.

Reply:
> But at what point is bullet https://github.com/dimagi/commcare-hq/pull/3 there cut out?

Yeah, that this doesn't add the case sharing groups at all, so the groups portion of the restore will differ.  This change only impacts `get_owner_ids` and thus the set of cases returned.

> But this couldn't possibly break any existing workflows since web users didn't have locations in the groups fixture before this anyways, right?

Yeah, though it could mean that existing apps that work with mobile workers won't work with web users (at least not until we give them groups, which I'd like to do at some point).